mission/simulator/platform-services-impl/src/main/java/esa/mo/platform/impl/provider/softsim/PowerControlSoftSimAdapter.java

```diff
@@ -44,70 +43,72 @@
   private final Map<Long, SimPayloadDevice> payloadIdByObjInstId;
   private static final Logger LOGGER = Logger.getLogger(PowerControlSoftSimAdapter.class.getName());
   private final ESASimulator instrumentsSimulator;
 
   public PowerControlSoftSimAdapter(ESASimulator instrumentsSimulator)
   {
     LOGGER.log(Level.INFO, "Initialisation");
-    this.instrumentsSimulator = instrumentsSimulator;
-    devices = new ArrayList<>();
-    deviceByName = new HashMap<>();
-    deviceByObjInstId = new HashMap<>();
+    this.instrumentsSimulator = instrumentsSimulator; //TODO this is never used?
+    deviceByType = new ConcurrentHashMap<>();
     payloadIdByObjInstId = new HashMap<>();
     initDevices();
   }
 
   private void initDevices()
   {
-    addDevice(new Device(true, 0L, new Identifier(
-        "Attitude Determination and Control System"), DeviceType.ADCS), SimPayloadDevice.FineADCS);
-    addDevice(new Device(true, 10L, new Identifier(
-        "Satellite Experimental Processing Platform 1"), DeviceType.OBC), SimPayloadDevice.SEPP1);
-    addDevice(new Device(true, 11L, new Identifier(
-        "Satellite Experimental Processing Platform 2"), DeviceType.OBC), SimPayloadDevice.SEPP2);
-    addDevice(new Device(false, 2L, new Identifier("S-Band Transceiver"), DeviceType.SBAND),
-        SimPayloadDevice.SBandTRX);
-    addDevice(new Device(false, 3L, new Identifier("X-Band Transmitter"), DeviceType.XBAND),
-        SimPayloadDevice.XBandTRX);
-    addDevice(new Device(false, 4L, new Identifier("Software Defined Radio"),
-        DeviceType.SDR), SimPayloadDevice.SDR);
-    addDevice(new Device(false, 5L, new Identifier("Optical Receiver"), DeviceType.OPTRX),
-        SimPayloadDevice.OpticalRX);
-    addDevice(new Device(false, 6L, new Identifier("HD Camera"), DeviceType.CAMERA),
-        SimPayloadDevice.HDCamera);
+    addDevice(new Device(true, 0L, SimPayloadDevice.FineADCS.identifier, DeviceType.ADCS), SimPayloadDevice.FineADCS);
+    addDevice(new Device(true, 10L, SimPayloadDevice.SEPP1.identifier, DeviceType.OBC), SimPayloadDevice.SEPP1);
+    addDevice(new Device(true, 11L, SimPayloadDevice.SEPP2.identifier, DeviceType.OBC), SimPayloadDevice.SEPP2);
+    addDevice(new Device(false, 2L, SimPayloadDevice.SBandTRX.identifier, DeviceType.SBAND),
+            SimPayloadDevice.SBandTRX);
+    addDevice(new Device(false, 3L, SimPayloadDevice.XBandTRX.identifier, DeviceType.XBAND),
+            SimPayloadDevice.XBandTRX);
+    addDevice(new Device(false, 4L, SimPayloadDevice.SDR.identifier,
+            DeviceType.SDR), SimPayloadDevice.SDR);
+    addDevice(new Device(false, 5L, SimPayloadDevice.OpticalRX.identifier, DeviceType.OPTRX),
+            SimPayloadDevice.OpticalRX);
+    addDevice(new Device(false, 6L, SimPayloadDevice.HDCamera.identifier, DeviceType.CAMERA),
+            SimPayloadDevice.HDCamera);
+    addDevice(new Device(true, 7L, SimPayloadDevice.GPS.identifier, DeviceType.GNSS),
```

Comment:
As we now track the statuses from PowerControlAdapter we should include GPS device here, right?
Wasn't sure where to check the device unitObjId - is 7 for GPS ok, or where can we check this?

Reply:
Correct. Lets track its status. By the way I think iADCS and GPS can be all marked as disabled by default now.
As for ObjId - as far as I can tell it is completely arbitrary number, as long as it is unique. I think SEPPs are offset by 10 because originally there was only 1 SEPP on the list with ID 1, and I did not want to shift IDs of all other payloads, so I simply made 11 out of 1, and added SEPP 2 with id 12.

Reply:
Done. GPS and ADCS marked as disabled by default.